fs/operations.go

```diff
@@ -646,15 +646,35 @@ func Check(fdst, fsrc Fs) error {
 // Lists in parallel which may get them out of order
 func ListFn(f Fs, fn func(Object)) error {
 	list := NewLister().SetFilter(Config.Filter).SetLevel(Config.MaxDepth).Start(f, "")
```

Comment:
It would probably make more sense to push the logic into the Lister object.

So change the SetError method to store the error and add a Error method to return it.